src/client/components/app/Transactions/components/TxError.tsx

```diff
@@ -1,62 +1,151 @@
-import { FC } from 'react';
+import { FC, useEffect, useState } from 'react';
 import styled from 'styled-components';
 
-import { Icon, Text, WarningIcon } from '@components/common';
+import { ChevronTxIcon, ErrorIcon, Icon, Text, WarningIcon } from '@components/common';
 import { useAppTranslation } from '@hooks';
 
+const StyledArrowIcon = styled(Icon)<{ isExpanded: boolean }>`
+  width: 1.6rem;
+  fill: currentColor;
+  flex-shrink: 0;
+  transform: rotate(180deg);
+  transition: transform 200ms ease-in-out;
+
+  ${({ isExpanded }) =>
+    isExpanded &&
+    `
+    transform: rotate(0);
+  `}
+`;
 const StyledIcon = styled(Icon)`
-  width: 4rem;
-  fill: ${({ theme }) => theme.colors.txModalColors.textContrast};
+  width: 2.4rem;
+  padding: 0 0.2rem;
+  fill: currentColor;
   flex-shrink: 0;
 `;
 
-const StyledText = styled(Text)`
+const StyledTextTitle = styled(Text)`
   color: inherit;
   max-width: 100%;
   overflow: hidden;
   text-overflow: ellipsis;
+  flex-grow: 1;
+  white-space: no-wrap;
+  line-height: 2.4rem;
+`;
+
+const StyledTextDescription = styled(Text)`
+  color: inherit;
+  max-width: 100%;
+  font-size: 1.2rem;
+  user-select: text;
 `;
 
 const StyledTxError = styled.div<{ errorType?: ErrorType }>`
   display: flex;
-  align-items: center;
-  justify-content: center;
-  gap: 5rem;
-  padding: 0 2.1rem;
-  font-weight: 500;
-  font-size: 1.4rem;
+  align-items: stretch;
+  font-weight: 100;
+  font-size: 1.6rem;
   border-radius: ${({ theme }) => theme.globalRadius};
-  text-transform: uppercase;
-  min-height: 7.8rem;
-  max-height: 10rem;
-  overflow: hidden;
-  overflow-y: auto;
+  padding: 0.8rem;
+  flex-direction: column;
+
+  background-color: ${({ theme }) => theme.colors.txModalColors.error.backgroundColor};
+  color: ${({ theme }) => theme.colors.txModalColors.error.color};
+
+  ${({ errorType, theme }) =>
+    errorType === 'warning' &&
+    `
+    background-color: ${theme.colors.txModalColors.warning.backgroundColor};
+    color: ${theme.colors.txModalColors.warning.color};
+  `}
+`;
+
+const StyledTxErrorTitle = styled.div`
+  display: flex;
   flex: 1;
+  gap: 0.8rem;
+  align-items: center;
 
-  background-color: ${({ theme }) => theme.colors.txModalColors.error};
-  color: ${({ theme }) => theme.colors.txModalColors.textContrast};
+  cursor: ${(props) => (props.onClick ? 'pointer' : 'default')};
+`;
+
+const StyledTxErrorDescription = styled.div<{ isExpanded: boolean; errorType?: ErrorType }>`
+  max-height: 0;
+  overflow-y: scroll;
+  transition: margin 200ms ease-in-out, max-height 200ms ease-in-out;
+
+  ${({ isExpanded }) =>
+    isExpanded &&
+    `
+    max-height: 8rem;
+    margin: .8rem 0;
+  `}
+
+  ::-webkit-scrollbar {
+    width: 0.2rem;
+  }
+
+  /* Track */
+  ::-webkit-scrollbar-track {
+    background: ${({ theme }) => theme.colors.txModalColors.error.color}80;
```

Comment:
typo

Reply:
nope, it adds 80% for opacity

Reply:
the thing is, what appens if the variable is an RGB, it will break, this only works with hexas

Reply:
should we rather set the opacity separately ?

Reply:
I can add a new constant to the theme just to have it as opacity, but there is no way to set opacity of the background separately

Reply:
you can always create a function to check if the color is an RGB, RGBA, or an hexa, and then apply the opacity depending that lol. 
I would just not use opacity in this case and not have problems in the future, and just try with another var. like mixing background color and surface color.